File: cpp/src/IceSSL/TraceLevels.cpp
// **********************************************************************
//
// Copyright (c) 2003-2005 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

#include <IceSSL/TraceLevels.h>
#include <Ice/Communicator.h>
#include <Ice/Properties.h>
#include <Ice/ProtocolPluginFacade.h>

using namespace std;
using namespace Ice;

void IceInternal::incRef(IceSSL::TraceLevels* p) { p->__incRef(); }
void IceInternal::decRef(IceSSL::TraceLevels* p) { p->__decRef(); }

IceSSL::TraceLevels::TraceLevels(const IceInternal::ProtocolPluginFacadePtr& protocolPluginFacade) :
    network(0),
    security(0),
    securityCat("Security")
{
    const_cast<int&>(network) = protocolPluginFacade->getNetworkTraceLevel();
    networkCat = protocolPluginFacade->getNetworkTraceCategory();

    PropertiesPtr properties = protocolPluginFacade->getCommunicator()->getProperties();
    const_cast<int&>(security) = properties->getPropertyAsInt("IceSSL.Trace.Security");
}

IceSSL::TraceLevels::~TraceLevels()
{
}

File: java/src/Freeze/src/main/java/Freeze/MapInternal/TraceLevels.java
// **********************************************************************
//
// Copyright (c) 2003-2017 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

package Freeze.MapInternal;

import Freeze.ConnectionI;

class TraceLevels
{
    TraceLevels(ConnectionI connection, String dbName)
    {
        errorPrefix = "Freeze DB DbEnv(\"" + connection.envName() + "\") Db(\"" + dbName + "\"): ";
        level = connection.trace();
        deadlockWarning = connection.deadlockWarning();
        logger = connection.getCommunicator().getLogger();
    }

    final String errorPrefix;
    final int level;
    final boolean deadlockWarning;
    final Ice.Logger logger;
}
